src/tir/schedule/traced_schedule.h

```diff
@@ -47,6 +47,15 @@ class TracedScheduleNode : public ConcreteScheduleNode {
 
  public:
   /******** Schedule: Sampling ********/
+  /*!
+   * \brief Sample an integer given the probability distribution
+   * \param candidates The candidates
+   * \param probs The probability distribution of the candidates
+   * \param decision The sampling decision
```

Comment:
Can you add documentation for `decision` saying what happens if it is set or not.

Reply:
+1 please document the behavior when decision is set

Reply:
Sure will do.